instance/models/load_balancer.py

```diff
@@ -210,13 +263,13 @@ def reconfigure(self, triggering_instance_id=None):
         reconfiguration of the load balancer.
         """
         self.logger.info("Reconfiguring load-balancing server %s", self.domain)
-        with self._configuration_lock():
-            self.run_playbook(self.get_ansible_vars(triggering_instance_id))
+        self.run_playbook(self.get_ansible_vars(triggering_instance_id))
 
     def deconfigure(self):
```

Comment:
Shouldn't this method also be using `@modifies_configuration` ?

Reply:
It did originally, but that doesn't really do what we want.  It try to grab the lock, and if it can't it would mark the LB as requiring _reconfiguration_, while it actually require _deconfiguration_.  This method is only called when an LB is deleted, and in this case the only reasonable behaviour is to block until we can grab the lock, and then deconfigure the LB.  Deferring the deconfiguration is not helpful.

Reply:
Nit: Ok, in that case, maybe the logic from `modifies_configuration` should be moved inside of `reconfigure(...)`? It would make the code easier to follow.

Reply:
Yep, I initially had it as a decorator to allow me to use it on both `reconfigure` and `deconfigure`, but now it makes more sense to be directly in `reconfigure`.